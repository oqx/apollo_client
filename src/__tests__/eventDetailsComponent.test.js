import React from "react";
import EventDetailsComponent from "../components/eventDetailsComponent";
import renderer from "react-test-renderer";

const event = {
  id: 243426,
  type: "Concert",
  date: null,
  artist: [
    { displayName: "RLGDPPL", id: 23431 },
    { displayName: "Joelletron", id: 23432 },
    { displayName: "Windmill", id: 23433 }
  ],
  venue: "First Avenue",
  radius: 5,
  latlng: [44.988767, -93.254362],
  distance: 3
};

describe("<EventDetailsComponent />", () => {
  function handleEventDetailsVisibility() {
    return true;
  }

  it("should render correctly", () => {
    const eventModal = renderer
      .create(
        <EventDetailsComponent
          handleVisibility={handleEventDetailsVisibility}
          event={event}
          eventDetailsIsOpen={true}
        />
      )
      .toJSON();
    expect(eventModal).toMatchSnapshot();
  });
});
