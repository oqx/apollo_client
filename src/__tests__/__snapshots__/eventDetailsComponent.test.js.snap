// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventDetailsComponent /> should render correctly 1`] = `
<section
  className="event-details "
>
  <div
    className="event-details__close"
    onClick={[Function]}
  >
    <i
      className="icon ion-android-close"
    />
  </div>
  <h3
    className="hd hd--3 hd--rule"
  >
    First Avenue
  </h3>
  <div
    className="event-details@geo"
  >
    <span>
      Begins 
      Invalid date
    </span>
      |
    <span
      className="event-details@geo__distance"
    >
      3
      mi
    </span>
    <a
      href="http://www.google.com/maps/place/44.988767,-93.254362"
      target="_blank"
    >
      <span
        className="ui-icon ui-icon--sm ui-icon--square ui-icon--tertiary"
        role="button"
      >
        <i
          className="icon ion-navigate"
        />
      </span>
    </a>
  </div>
  <div
    className="event-details__info"
  >
    <div
      className=""
    >
      Artists
    </div>
    <div
      className="event-details__col-date"
    >
      <time
        className="event-details__date"
        dateTime="{moment(event.date).format()}"
      >
        
      </time>
    </div>
  </div>
  <ul
    className="event-details@list"
  >
    <li
      className="event-details@list__item"
    >
      RLGDPPL
    </li>
    <li
      className="event-details@list__item"
    >
      Joelletron
    </li>
    <li
      className="event-details@list__item"
    >
      Windmill
    </li>
  </ul>
</section>
`;
